collector/info_schema_innodb_sys_tablespaces.go

```diff
@@ -88,14 +105,24 @@ func (ScrapeInfoSchemaInnodbTablespaces) Scrape(ctx context.Context, instance *i
 	var tablespacesTablename string
 	var query string
 	db := instance.getDB()
+
 	err := db.QueryRowContext(ctx, innodbTablespacesTablenameQuery).Scan(&tablespacesTablename)
 	if err != nil {
 		return err
 	}
 
 	switch tablespacesTablename {
 	case "INNODB_SYS_TABLESPACES", "INNODB_TABLESPACES":
-		query = fmt.Sprintf(innodbTablespacesQuery, tablespacesTablename, tablespacesTablename)
+		if instance.flavor == FlavorMariaDB {
+			if instance.version.GTE(semver.MustParse("10.5.0")) {
+				query = fmt.Sprintf(innodbTablespacesQueryMariaDB, tablespacesTablename, tablespacesTablename)
+			} else {
+				query = fmt.Sprintf(innodbTablespacesQueryMySQL, tablespacesTablename, tablespacesTablename)
+			}
+		}
+		if instance.flavor == FlavorMySQL {
+			query = fmt.Sprintf(innodbTablespacesQueryMySQL, tablespacesTablename, tablespacesTablename)
+		}
```

Comment:
This has a chance of producing an empty/incorrect `query` value. I'd suggest this logic to prevent no matching of `flavor`.

```suggestion
		query = fmt.Sprintf(innodbTablespacesQueryMySQL, tablespacesTablename, tablespacesTablename)
		if instance.flavor == FlavorMariaDB {
			if instance.version.GTE(semver.MustParse("10.5.0")) {
				query = fmt.Sprintf(innodbTablespacesQueryMariaDB, tablespacesTablename, tablespacesTablename)
			}
		}
```